upgrade/pack_builder/SupportedTargets.cpp

```diff
@@ -14,38 +14,63 @@ namespace application
         return *this;
     }
 
+    SupportedTargetsBuilder& SupportedTargetsBuilder::Order(uint8_t order)
+    {
+        this->order.Emplace(order);
+        return *this;
+    }
+
     SupportedTargetsBuilder& SupportedTargetsBuilder::AddCmd(const SupportedTargets::Target& target)
     {
         AddToMandatoryWhenNecessary(target);
+        AddInOrder(target);
         targets.cmd.emplace_back(target);
         return *this;
     }
 
     SupportedTargetsBuilder& SupportedTargetsBuilder::AddHex(const SupportedTargets::Target& target)
     {
         AddToMandatoryWhenNecessary(target);
+        AddInOrder(target);
         targets.hex.emplace_back(target);
         return *this;
     }
 
     SupportedTargetsBuilder& SupportedTargetsBuilder::AddElf(const SupportedTargets::Target& target, uint32_t offset)
     {
         AddToMandatoryWhenNecessary(target);
+        AddInOrder(target);
         targets.elf.emplace_back(target, offset);
         return *this;
     }
 
     SupportedTargetsBuilder& SupportedTargetsBuilder::AddBin(const SupportedTargets::Target& target, uint32_t offset)
     {
         AddToMandatoryWhenNecessary(target);
+        AddInOrder(target);
         targets.bin.emplace_back(target, offset);
         return *this;
     }
 
     void SupportedTargetsBuilder::AddToMandatoryWhenNecessary(const SupportedTargets::Target& target)
     {
         if (mandatory)
+        {
             targets.mandatory.emplace_back(target);
+            mandatory = false;
+        }
+    }
+
+    void SupportedTargetsBuilder::AddInOrder(const SupportedTargets::Target& target)
+    {
+        if (order)
+        {
+            if (targets.order.size() < *order)
```

Comment:
The more proper type here would be a map iso a vector, e.g. std::map<uint8_t, std::vector<Target>> so that you don't have to resize for unused orders. You can just do 
`targets.order[*order].emplace_back(target);`